### ISSUE TITLE
AuthResult.Status fields are inaccessible

### ISSUE THREAD

**Issue by halestock** [okta/oktasdk-java#54]
The various status strings in AuthResult.Status cannot be referenced as they are non-static and cannot be referenced from a static context (e.g. directly as AuthResult.Status.SUCCESS). In order to be used, a new instance of AuthResult.Status would have to be instantiated.

**Comment by jmelberg-okta** [comment]
@halestock: To keep with the coding conventions throughout this SDK, we suggest checking the `AuthResult` status with the `getStatus()` method. For example:
```java
AuthApiClient authClient = new AuthApiClient(oktaSettings);

// Check if the user credentials are valid
AuthResult result = authClient.authenticate(username, password, someRelayState);

// Verify SUCCESS status
if (result.getStatus().equals("SUCCESS")) { /* Do something */ }
```

**Comment by halestock** [comment]
@jmelberg-okta I'm referring to the fact that you cannot use the fields in AuthResult.Status to compare against when calling `result.getStatus()`. Ideally, you would use something like `result.getStatus().equals(AuthResult.Status.SUCCESS)` instead of hardcoding the values in my app. However, since they are not static, I would instead have to do:
`
AuthResult result = authClient.authenticate(username, password, someRelayState);
AuthResult.Status statuses = new AuthResult.Status();
// verify
if (result.getStatus().equals(statuses.SUCCESS)) { /* do something */ }
`

This seems to be pretty unnecessary.

**Comment by jmelberg-okta** [comment]
@halestock: Good point. I have created a ticket with our team to update the SDK.

If you are interested in pushing this change yourself, please visit [developer.okta.com/cla](http://developer.okta.com/cla) for more information.

Thanks!

**Comment by jmelberg-okta** [comment]
Closing as was implemented with PR [#58](https://github.com/okta/oktasdk-java/pull/58)
